sklearn/impute/_iterative.py

```diff
@@ -558,8 +558,14 @@ def fit_transform(self, X, y=None):
         else:
             self._estimator = clone(self.estimator)
 
-        if hasattr(self._estimator, 'random_state'):
-            self._estimator.random_state = self.random_state_
+        if (
+            hasattr(self._estimator, "random_state")
+            and self._estimator.random_state is None
+            and self.random_state is not None
+        ):
+            random_state = self.random_state_.randint(
+                np.iinfo(np.int32).max)
+            self._estimator.set_params(random_state=random_state)
```

Comment:
This should simply be:

``` python
             self._estimator.set_params(random_state=self.random_state)
```

If the given estimator doesn't support int or rng or None, then user shouldn't pass that as the random state.